web/routing.go

```diff
@@ -79,9 +79,11 @@ func SetupAppsHandler(appsHandler echo.HandlerFunc) echo.HandlerFunc {
 		middlewares.Accept(middlewares.AcceptOptions{
 			DefaultContentTypeOffer: echo.MIMETextHTML,
 		}),
+		middlewares.NeedFlags,
 		middlewares.CheckInstanceBlocked,
 		middlewares.CheckInstanceDeleting,
 		middlewares.CheckTOSDeadlineExpired,
+		middlewares.CheckOAuthClientsLimitExceeded,
```

Comment:
I wouldn't have done that with a middleware:

1. I would prefer to do the check/redirection only for the HTML pages, not for the assets (JS, CSS, images)
2. I don't think it is a good idea to do that for people going to a shared by link page, as it can confuse them by saying that they have reached the limit of the number of their devices, but it's not their devices, but the devices of the owner of the instance with the sharing that is counted.

Reply:
Hum I tried to copy what was done for the ToS blocker as it seemed very similar.
I guess this check should be moved to the `Serve` method of the `apps` package?!